### ISSUE TITLE
Not able to Load Index.html From Local URL present in Xcode project

### ISSUE THREAD

**Issue by ShashankOodles1** [tradingview/charting-library-examples#213]
Hi Team,
We are currently facing issue when we try to load URL (Index.htlm) from Xcode project locally.
we are able to get URL (Load from Bundle) but the screen is blank.

but as in the demo URL i am able to load the chart. 

Here is my Code 

```swift
class ViewController: UIViewController {
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        let frame = UIScreen.main.bounds;
        
        let configuration = WKWebViewConfiguration()
        configuration.preferences.setValue(true, forKey: "allowFileAccessFromFileURLs")
        configuration.preferences.javaScriptEnabled = true
        let urlRaw = Bundle.main.url(forResource: "index", withExtension: "html")
        
        let webView = WKWebView(frame: frame, configuration: configuration)
        guard let url = urlRaw else {
            print("No file at url")
            return
        }
        print(url)
        webView.loadFileURL(url, allowingReadAccessTo: url)
        
        view.addSubview(webView)
    }
    
    
}
```

Here is my Index.html File :
```html
<!DOCTYPE html>
<html>
	<head>

		<title>TradingView Charting Library demo</title>

		<!-- Fix for iOS Safari zooming bug -->
		<meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0,minimum-scale=1.0">
		<meta http-equiv="X-UA-Compatible" content="IE=Edge">

		<script type="text/javascript" src="charting_library/charting_library.standalone.js"></script>
		<script type="text/javascript" src="datafeeds/udf/dist/polyfills.js"></script>
		<script type="text/javascript" src="datafeeds/udf/dist/bundle.js"></script>

		<script type="text/javascript">

			function getParameterByName(name) {
				name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
				var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
					results = regex.exec(location.search);
				return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
			}

			function initOnReady() {
				var widget = window.tvWidget = new TradingView.widget({
					// debug: true, // uncomment this line to see Library errors and warnings in the console
					fullscreen: true,
					symbol: 'AAPL',
					interval: '1D',
					container_id: "tv_chart_container",

					//	BEWARE: no trailing slash is expected in feed URL
					datafeed: new Datafeeds.UDFCompatibleDatafeed("https://demo_feed.tradingview.com"),
					library_path: "charting_library/",
					locale: getParameterByName('lang') || "en",

					disabled_features: ["use_localstorage_for_settings"],
					enabled_features: ["study_templates"],
					charts_storage_url: 'https://saveload.tradingview.com',
					charts_storage_api_version: "1.1",
					client_id: 'tradingview.com',
					user_id: 'public_user_id',
					theme: getParameterByName('theme'),
				});
			};

			window.addEventListener('DOMContentLoaded', initOnReady, false);
		</script>

	</head>

	<body style="margin:0px;">
		<div id="tv_chart_container"></div>
	</body>

</html>
```

Please let me know what i am doing wrong and share me the possible solution

**Comment by timocov** [comment]
@makedonsky94 can you please take a look?

**Comment by makedonsky94** [comment]
I guess there an issue with SSL certificates as we faced on iOS devices.

Replace 
```swift
datafeed: new Datafeeds.UDFCompatibleDatafeed("https://demo_feed.tradingview.com")
```
with
```swift
datafeed: new Datafeeds.UDFCompatibleDatafeed("https://demo-feed-data.tradingview.com")
```

**Comment by timocov** [comment]
https://bugs.webkit.org/show_bug.cgi?id=221953

**Comment by ShashankOodles1** [comment]
Hi Alexandr
I have changed the Datafeed url but still blank screen Coming no error!
Please Provide me any link for implementing trading chart view in iOS.


Thanks

**Comment by makedonsky94** [comment]
Can you check the error in Safari Web Inspector? 

After the first open, Web Inspector won't have any error messages. You should click on reload button

**Comment by ShashankOodles1** [comment]
Hi Team
I Tried Everything Getting this message
Example[6375:237761] [Process] 0x7fc89701be20 - [pageProxyID=6, webPageID=7, PID=6382] WebPageProxy::didFailProvisionalLoadForFrame: frameID = 3, domain = NSURLErrorDomain, code = -1100
please Help Team

**Comment by veerChauhan** [comment]
```
<!DOCTYPE html>
<html>
    <head>
        
        <title>TradingView Charting Library demo</title>
        
        <!-- Fix for iOS Safari zooming bug -->
        <meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0,minimum-scale=1.0">
            <meta http-equiv="X-UA-Compatible" content="IE=Edge">
                <h1>Exchange Heading</h1>
                <p>Exchanege Paragraph</p>
                </head>
    
    <body style="margin:0px;">
        <div id="tv_chart_container"></div>
    </body>
    
</html>


<!---->
<!--<!DOCTYPE html>-->
<!--<html>-->
<!--	<head>-->
<!---->
<!--		<title>TradingView Charting Library demo</title>-->
<!---->
<!--		<!-- Fix for iOS Safari zooming bug -->-->
<!--		<meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0,minimum-scale=1.0">-->
<!--		<meta http-equiv="X-UA-Compatible" content="IE=Edge">-->
<!---->
<!--		<script type="text/javascript" src="charting_library/charting_library.standalone.js"></script>-->
<!--		<script type="text/javascript" src="datafeeds/udf/dist/polyfills.js"></script>-->
<!--		<script type="text/javascript" src="datafeeds/udf/dist/bundle.js"></script>-->
<!---->
<!--		<script type="text/javascript">-->
<!---->
<!--			function getParameterByName(name) {-->
<!--				name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");-->
<!--				var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),-->
<!--					results = regex.exec(location.search);-->
<!--				return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));-->
<!--			}-->
<!---->
<!--			function initOnReady() {-->
<!--				var widget = window.tvWidget = new TradingView.widget({-->
<!--					// debug: true, // uncomment this line to see Library errors and warnings in the console-->
<!--                    debug: true,-->
<!--					fullscreen: true,-->
<!--					symbol: 'AAPL',-->
<!--					interval: '1D',-->
<!--					container_id: "tv_chart_container",-->
<!---->
<!--					//	BEWARE: no trailing slash is expected in feed URL-->
<!--					datafeed: new Datafeeds.UDFCompatibleDatafeed("demo-feed-data.tradingview.com"),-->
<!--					library_path: "charting_library",-->
<!--					locale: getParameterByName('lang') || "en",-->
<!---->
<!--					disabled_features: ["use_localstorage_for_settings"],-->
<!--					enabled_features: ["study_templates"],-->
<!--					charts_storage_url: 'https://saveload.tradingview.com',-->
<!--					charts_storage_api_version: "1.1",-->
<!--					client_id: 'tradingview.com',-->
<!--					user_id: 'public_user_id',-->
<!--					theme: getParameterByName('theme'),-->
<!--				});-->
<!--			};-->
<!---->
<!--			window.addEventListener('DOMContentLoaded', initOnReady, false);-->
<!--		</script>-->
<!---->
<!--	</head>-->
<!---->
<!--	<body style="margin:0px;">-->
<!--		<div id="tv_chart_container"></div>-->
<!--	</body>-->
<!---->
<!--</html>-->
```
We are able to load other URL eg. the webpage where we have implemented trading view as well even this above code of heading and para works fine but the comented code is not working.

**Comment by makedonsky94** [comment]
@veerChauhan I found that the code from your example does not working properly. 

There you need to change 

`library_path: "charting_library"` to `library_path: "charting_library/"`

**Comment by makedonsky94** [comment]
@ShashankOodles1 That means there something bad happened in requests to local files. But I can't reproduce the bug with your example. Guess I'm using the different version of charting library. I need the error from Web Inspector. You can find it in Safari browser
![image](https://user-images.githubusercontent.com/11327827/109008804-cff90780-76be-11eb-94f2-baf25369a3d5.png)

**Comment by ShashankOodles1** [comment]
@makedonsky94 Do have any demo for implementation of ios trading chart view.

**Comment by makedonsky94** [comment]
https://github.com/tradingview/charting-library-examples/tree/master/ios-swift
It was tested with the last version of charting library. If you have any troubles with this example please let us know what errors do you have in the Safari Web Inspector

**Comment by ShashankOodles1** [comment]
![Screenshot 2021-03-02 at 5 04 34 PM](https://user-images.githubusercontent.com/74229767/109642750-6161f100-7b79-11eb-96b7-ff199f6f0af7.png)
<!DOCTYPE HTML>
<html>
	<head>

		<title>TradingView Charting Library demo</title>

		
		<meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0,minimum-scale=1.0">
		<meta http-equiv="X-UA-Compatible" content="IE=Edge">

		<script type="text/javascript" src="charting_library.standalone.js"></script>
		<script type="text/javascript" src="datafeeds/udf/dist/polyfills.js"></script>
		<script type="text/javascript" src="datafeeds/udf/dist/bundle.js"></script>

		<script type="text/javascript">
        //    var TradingView;
			function getParameterByName(name) {
               /// TradingView = "charting_library.standalone.js"
				name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
				var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
					results = regex.exec(location.search);
				return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
			}
         
			function initOnReady() {
                
                const widgetOptions = ({
					// debug: true, // uncomment this line to see Library errors and warnings in the console
					fullscreen: true,
					symbol: 'AAPL',
					interval: '1D',
					container_id: "tv_chart_container",

					//	BEWARE: no trailing slash is expected in feed URL
					datafeed: new Datafeeds.UDFCompatibleDatafeed("https://demo_feed_data.tradingview.com"),
					library_path: "charting_library/",
					locale: getParameterByName('lang') || "en",

					disabled_features: ["use_localstorage_for_settings"],
					enabled_features: ["study_templates"],
					charts_storage_url: 'https://saveload.tradingview.com',
					charts_storage_api_version: "1.1",
					client_id: 'tradingview.com',
					user_id: 'public_user_id',
					theme: getParameterByName('theme'),
				});
			};

			window.addEventListener('DOMContentLoaded', initOnReady, false);
		</script>

	</head>

	<body style="margin:0px;">
		<div id="tv_chart_container"></div>
	</body>

</html>

**Comment by makedonsky94** [comment]
```
<script type="text/javascript" src="charting_library.standalone.js"></script>
```
In which folder the js file do you have? As I can see in the previous code example you had `src="charting_library/charting_library.standalone.js"`

Also, you have incorrect domain name in datafeed url.
Replace 
```
datafeed: new Datafeeds.UDFCompatibleDatafeed("https://demo_feed_data.tradingview.com"),
```
with
```
datafeed: new Datafeeds.UDFCompatibleDatafeed("https://demo-feed-data.tradingview.com"),
```

In the row 
```
const widgetOptions = ({
```
you should call 
```
const widgetOptions = window.tvWidget = new TradingView.widget({
```

Please, check your code and try to run

**Comment by ShashankOodles1** [comment]
![Screenshot 2021-03-02 at 9 05 53 PM](https://user-images.githubusercontent.com/74229767/109672638-38eaee80-7b9b-11eb-8921-bbcc25a3abd9.png)
![Screenshot 2021-03-02 at 9 07 14 PM](https://user-images.githubusercontent.com/74229767/109672711-4a33fb00-7b9b-11eb-9aae-50f5487836ad.png)
<!DOCTYPE HTML>
<html>
	<head>

		<title>TradingView Charting Library demo</title>

		
		<meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0,minimum-scale=1.0">
		<meta http-equiv="X-UA-Compatible" content="IE=Edge">

		<script type="text/javascript" src="charting_library.standalone.js"></script>
		<script type="text/javascript" src="datafeeds/udf/dist/polyfills.js"></script>
		<script type="text/javascript" src="datafeeds/udf/dist/bundle.js"></script>

		<script type="text/javascript">
        //    var TradingView;
			function getParameterByName(name) {
               /// TradingView = "charting_library.standalone.js"
				name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
				var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
					results = regex.exec(location.search);
				return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
			}
         
			function initOnReady() {
                
                const widgetOptions = window.tvWidget = new TradingView.widget({
					// debug: true, // uncomment this line to see Library errors and warnings in the console
					fullscreen: true,
					symbol: 'AAPL',
					interval: '1D',
					container_id: "tv_chart_container",

					//	BEWARE: no trailing slash is expected in feed URL
					datafeed: new Datafeeds.UDFCompatibleDatafeed("https://demo-feed-data.tradingview.com"),
					library_path: "charting_library/",
					locale: getParameterByName('lang') || "en",

					disabled_features: ["use_localstorage_for_settings"],
					enabled_features: ["study_templates"],
					charts_storage_url: 'https://saveload.tradingview.com',
					charts_storage_api_version: "1.1",
					client_id: 'tradingview.com',
					user_id: 'public_user_id',
					theme: getParameterByName('theme'),
				});
			};

			window.addEventListener('DOMContentLoaded', initOnReady, false);
		</script>

	</head>

	<body style="margin:0px;">
		<div id="tv_chart_container"></div>
	</body>

</html>
I followed what you said now i have send you everything please check

**Comment by ShashankOodles1** [comment]
@makedonsky94  please see and help

**Comment by makedonsky94** [comment]
Yep, the document can't load js library. So you got that error

Replace
```
	<script type="text/javascript" src="charting_library.standalone.js"></script>
```
with
```
	<script type="text/javascript" src="charting_library/charting_library.standalone.js"></script>
```

**Comment by ShashankOodles1** [comment]
![Screenshot 2021-03-02 at 9 17 48 PM](https://user-images.githubusercontent.com/74229767/109674283-ced34900-7b9c-11eb-9a95-4ebed3ea02f0.png)

**Comment by ShashankOodles1** [comment]
![Screenshot 2021-03-02 at 9 19 18 PM](https://user-images.githubusercontent.com/74229767/109674497-f9250680-7b9c-11eb-8208-edb2e754f95e.png)

**Comment by ShashankOodles1** [comment]
![Screenshot 2021-03-02 at 9 20 37 PM](https://user-images.githubusercontent.com/74229767/109674819-3c7f7500-7b9d-11eb-8d8c-114dc64d7ce8.png)
in my charting Library i have two files same name so i used legacy system does it change cause any error
![Screenshot 2021-03-02 at 9 21 04 PM](https://user-images.githubusercontent.com/74229767/109675002-65076f00-7b9d-11eb-9b27-b8b8e06eea52.png)

**Comment by ShashankOodles1** [comment]
file:///Users/shashankmishra/Desktop/ios-swift/App/ChartingLibrary/charting_library/index.html this is the url when i drag my index file to safari browser

**Comment by makedonsky94** [comment]
Please, check your bundles. You should keep nesting of folders
![image](https://user-images.githubusercontent.com/11327827/109677192-7abf6900-7b8a-11eb-8116-bfee1c6e8444.png)

**Comment by ShashankOodles1** [comment]
![Screenshot 2021-03-02 at 10 37 53 PM](https://user-images.githubusercontent.com/74229767/109686096-f845a200-7ba7-11eb-9f14-aa559c114067.png)
still getting this

**Comment by ShashankOodles1** [comment]
![Uploading Screenshot 2021-03-02 at 10.39.37 PM.png…]()

**Comment by ShashankOodles1** [comment]
![Screenshot 2021-03-02 at 10 41 38 PM](https://user-images.githubusercontent.com/74229767/109686700-8457c980-7ba8-11eb-880e-dccbbcdf56fc.png)
![Screenshot 2021-03-02 at 10 41 50 PM](https://user-images.githubusercontent.com/74229767/109686714-8883e700-7ba8-11eb-809b-6c2df74dd895.png)

**Comment by makedonsky94** [comment]
https://github.com/tradingview/charting-library-examples/issues/213#issuecomment-789006933
At this screenshot the library loaded successfully. You broke something. Check the paths carefully

**Comment by ShashankOodles1** [comment]
@makedonsky94  chart is coming on safari but won't able to run on device

**Comment by ShashankOodles1** [comment]
![Screenshot 2021-03-02 at 11 57 08 PM](https://user-images.githubusercontent.com/74229767/109696433-0f3dc180-7bb3-11eb-8947-7ff4f4d5ab3d.png)
this is coming in safari

**Comment by ShashankOodles1** [comment]
Hi @makedonsky94  i have implemented trading chart in Local Project but i am not able to change symbol as it is not accessible form

**Comment by neerajk207** [comment]
> Hi @makedonsky94 i have implemented trading chart in Local Project but i am not able to change symbol as it is not accessible form function initOnReady() { const widgetOptions = window.tvWidget = new TradingView.widget({ // debug: true, // uncomment this line to see Library errors and warnings in the console fullscreen: true, symbol: '14', interval: '1D', container_id: "tv_chart_container", so can you let me know how i can make symbol accessible and make dynamic

@ShashankOodles1  were you able to make the symbol dynamic? if yes, please let me know.

**Comment by ShashankOodles1** [comment]
yes @neerajk207  you just have replace symbol in index file by sending via webview property let share you screen shot and let me if issue still persists 
note :- selected chart id which is dynamic on your screen
![Screenshot 2022-03-30 at 12 52 25 AM](https://user-images.githubusercontent.com/74229767/160691040-25cb5465-430b-48f6-98c6-45e6734a15bb.png)
ersist
![Screenshot 2022-03-30 at 12 51 40 AM](https://user-images.githubusercontent.com/74229767/160690906-7f065a41-fd74-4327-926a-3e0f58c7141b.png)
s

**Comment by salman-ar-sar** [comment]
@timocov @ShashankOodles1 I am working on integrating TradingView into my react native app. I have put the library into it. It is working fine in android and iOS 15 and above. But on iOS 14 I am getting a blank screen with no error in metro.
When I tried to debug using Safari inspector I am getting this error:
`Failed to load resource: The operation couldn’t be completed. (WebKitBlobResource error 1.)`
Any idea what could be the issue and how to solve it?

**Comment by joshi-zanskar** [comment]
Hey

<img width="352" alt="Image" src="https://github.com/user-attachments/assets/ba218ede-7935-4669-8630-d8e797811139" />

 Team,

I am experiencing an issue while using Xcode 16. When attempting to add resources to "Copy Bundle Resources," the process only adds individual files rather than the entire folder. This behavior is preventing me from loading the charting library correctly.

I would greatly appreciate any guidance on resolving this issue.

<img width="1817" alt="Image" src="https://github.com/user-attachments/assets/77845fae-b3b3-4b55-a5ab-d55119e99f32" />

**Comment by romfrancois** [comment]
Have you tried adding all files individually by any chance?

**Comment by joshi-zanskar** [comment]
Thanks @romfrancois 

I successfully resolved the issue by manually adding the reference as follows:

Navigate to Project -> Target -> Build Phase -> Copy Bundle Resources.

Click the '**+**' button to add the necessary files.

Set the following options:
Destination: Copy items if needed
Added Folders: Create Folder references

Note:  I am using React Native, so I passed the reference from my React Native src folder.
